### ISSUE TITLE
img2pdf?

### ISSUE THREAD

**Issue by mara004** [stefan6419846/image-to-pdf#4]
Just wondering whether you are aware of [img2pdf](https://gitlab.mister-muffin.de/josch/img2pdf/)?
It's always lossless, supports alpha channels through `SMask`s, and can mostly avoid CPU-hungry re-encoding.
I think it's probably the best OSS tool in existence for this task (although I might have a few code style / design decision disagreements).

**Comment by stefan6419846** [comment]
Thanks for opening this issue.

I am aware of `img2pdf` and it is being used in the same project, but for some reasons (probably historical) this specific task had been implemented using *Pillow* back then. If I remember correctly, `img2pdf` would have its own limitations/issues, although I am not completely sure out of my head without digging further. (I just stumbled upon the required dependency on `pikepdf`, although it being optional internally.)

In this specific case, I needed a quick solution and focused on the *Pillow* issue linked in the README. In the future, this might change and I re-evaluate `img2pdf` for the corresponding use-cases, but for now I will probably stay with the current approach and this package.

**Comment by mara004** [comment]
Yeah, historically, img2pdf did not support alpha channels. It was contributed by Tamas Zahola some time ago.

Current limitations I'm aware of is lack of support for >8bpp TIFFs, and >8bpp PNGs with alpha channel. Due to issues in Pillow, these images would be implicitly truncated to 8 bpp, and the author strictly refuses the idea of adding an option to let them pass through lossily.
